pkg/operator/ceph/object/rgw.go

```diff
@@ -133,54 +133,48 @@ func (c *clusterConfig) startRGWPods() error {
 				logger.Info("setting rgw config flags")
 				err = c.setDefaultFlagsMonConfigStore(rgwConfig.ResourceName)
 				if err != nil {
-					return errors.Wrapf(err, "failed to set default rgw config options")
+					return errors.Wrap(err, "failed to set default rgw config options")
 				}
 			}
 		}
 
 		// Create deployment
 		deployment := c.createDeployment(rgwConfig)
-		logger.Infof("object store %s deployment %s started", c.store.Name, deployment.Name)
+		logger.Infof("object store %q deployment %q started", c.store.Name, deployment.Name)
+
+		// Set owner ref to cephObjectStore object
+		err = controllerutil.SetControllerReference(c.store, deployment, c.scheme)
+		if err != nil {
+			return errors.Wrapf(err, "failed to set owner reference for ceph object user %q secret", deployment.Name)
```

Comment:
If we fail to set a controller reference what if we logged the error and continued with the reconcile? It doesn't seem fatal.

Reply:
Not fatal but the resource won't be garbage collected once the CR goes away :-(

Reply:
And this is not the desired state we want